core/templates/pages/exploration-editor-page/services/change-list.service.ts

```diff
@@ -100,18 +102,22 @@ export class ChangeListService implements OnInit {
     // If error is present -> Check for the type of error occurred
     // (Display the corresponding modals in both cases, if not already
     // opened):
-    // - Version Mismatch.
+    // - Changes are not mergeable when a version mismatch occurs.
     // - Non-strict Validation Fail.
     this.explorationDataService.autosaveChangeListAsync(
       explorationChangeList,
       response => {
-        if (!response.is_version_of_draft_valid) {
+        if (!response.changes_are_mergeable) {
           if (!this.autosaveInfoModalsService.isModalOpen()) {
             this.autosaveInfoModalsService.showVersionMismatchModal(
               explorationChangeList);
           }
         }
         this.autosaveInProgressEventEmitter.emit(false);
+        if (!response.is_version_of_draft_valid &&
+          response.changes_are_mergeable) {
+          this.windowRef.nativeWindow.location.reload();
+        }
```

Comment:
When changes are mergeable we are expecting the version_of_draft to be valid, right?

Reply:
Yes, but we kept this check of is_version_of_draft_valid because changes_are_mergeable even if the versions are same and we don't need a reload when the versions are the same. That's why I used this.

Reply:
In that case, can we change `is_version_of_draft_valid` to `version_of_draft_updated`? (Which will help us know whether in the last autoSave call the version of the draft was updated?)

Reply:
But we don't want to check that whether the version of the draft is updated or not, we are just checking that at present is the version in the frontend same as the version in the backend, therefore I guess the name which we are using at present is suitable for it.